marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/models/createModel.sjs

```diff
@@ -46,10 +49,26 @@ model.definitions[name] = {
   properties: {}
 };
 
+if(namespace || namespacePrefix){
+  if(!namespace){
+    httpUtils.throwBadRequest(`You cannot enter a prefix without specifying a namespace URI `);
+  }
+  if(!namespacePrefix){
+    httpUtils.throwBadRequest(`Since you entered a namespace, you must specify a prefix.`);
+  }
+  model.definitions[name].namespace = namespace;
```

Comment:
This assumption - that the namespace/namespacePrefix are to be applied to the primary entity type - seems reasonable to do. But it raises the question of what to do about structured type properties.